app/src/main/java/org/mozilla/focus/settings/ExperimentsSettingsFragment.kt

```diff
@@ -9,23 +9,37 @@ import android.os.Bundle
 import android.support.v14.preference.SwitchPreference
 import android.support.v7.preference.PreferenceFragmentCompat
 import com.jakewharton.processphoenix.ProcessPhoenix
+import mozilla.components.service.fretboard.ExperimentDescriptor
+import mozilla.components.service.fretboard.Fretboard
+import org.mozilla.focus.FocusApplication
 import org.mozilla.focus.R
 import org.mozilla.focus.web.Config
 import org.mozilla.focus.web.ENGINE_PREF_STRING_KEY
 
+const val EXPERIMENTS_JSON_FILENAME = "experiments.json"
+const val EXPERIMENTS_BASE_URL = "https://settings.prod.mozaws.net/v1"
+const val EXPERIMENTS_BUCKET_NAME = "main"
+const val EXPERIMENTS_COLLECTION_NAME = "focus-experiments"
+
+val experimentDescriptor = ExperimentDescriptor(Config.EXPERIMENT_DESCRIPTOR_GECKOVIEW_ENGINE)
+
 class ExperimentsSettingsFragment : PreferenceFragmentCompat(), SharedPreferences.OnSharedPreferenceChangeListener {
     companion object {
         const val FRAGMENT_TAG = "ExperimentSettings"
     }
 
     private var rendererPreferenceChanged = false
+    private lateinit var fretboard: Fretboard
+    private lateinit var enginePreference: SwitchPreference
+    lateinit var application: FocusApplication
 
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         addPreferencesFromResource(R.xml.experiments_settings)
-        val enginePref: SwitchPreference? = preferenceManager
-                .findPreference(ENGINE_PREF_STRING_KEY) as SwitchPreference?
-        enginePref?.isChecked = preferenceManager.sharedPreferences
-                .getBoolean(ENGINE_PREF_STRING_KEY, Config.DEFAULT_NEW_RENDERER)
+        enginePreference = preferenceManager
+                .findPreference(ENGINE_PREF_STRING_KEY) as SwitchPreference
+        application = activity?.application as FocusApplication
+        fretboard = application.fretboard
+        enginePreference.isChecked = fretboard.isInExperiment(application, experimentDescriptor)
```

Comment:
nit: Maybe we could make Fretboard (or a an isInExperiments method?) available via an extension method?

Maybe something like:

```Kotlin
val Context.application: FocusApplication
    get() = applicationContext as FocusApplication

val Context.fretboard: Fretboard
    get() = application.fretboard
```

Or maybe additionally:

```Kotlin
fun Context.isInExperiment(descriptor: ExperimentDescriptor): Boolean =
    fretboard.isInExperiment(this, experimentDescriptor)
```

What do you think?

Reply:
That works for me.